### DISCUSSION TITLE
Cells splitting into multiple pieces during simulation

### DISCUSSION THREAD

**Discussion by neonine2** [ingewortel/artistoo#12] [Q&A] [answered]
While playing around with different simulations, I have ran into situations where when I try to mix and match default parameters from different simulations, I see cells splitting into multiple pieces. I am wondering what the sensitivity of this behavior to various parameters, and if there are just general tips on how to avoid this type of behavior when setting up a simulation. Thank you.

**Comment by ingewortel** [comment, marked as answer]
Hi!

Cell breaking is indeed an artefact you may get  when the “forces” emerging from the energy terms in deltaH are not well-balanced. If cells split often, you can try the following things to solve it:

1. **Tune parameters, starting with the Adhesion and (if present) the PerimeterConstraint.** 
These terms in deltaH are most important in keeping the cell intact; both penalize large “interfaces” between a cell and its outside world, which cells can minimize by staying intact. Tuning the (lambda and other) parameters of these terms may improve cell integrity. Note that the required parameter values may change somewhat depending on:

    - _The other deltaH terms you have_. 
    What matters for P_copy is the total deltaH, so if you add a new term that “overshadows” the adhesion/perimeter terms in deltaH, they may no longer work in keeping the cell whole (even when the same values did work in another simulation). E.g. when adding the ActivityConstraint, in some cases the positive feedback may result in such large energies that the deltaH_act will overshadow the deltaH_adhesion and deltaH_perimeter, after which the cell may break. So the same _absolute_ value of a deltaH term may have a different effect depending on the size of the other terms, because their _relative_ "weight" changes.
    - _The environment_. 
    For example, which target perimeter you should choose for a certain cell volume may depend on the geometry (e.g. 2D cells have a lower surface to volume ratio than 3D ones, as do  roundish, “freely” moving cells compared to elongated ones constrained in microchannels).

    Keep in mind though that cell breaking is not _always_ an artefact; real cells may also break when they are subject to extremely high forces. E.g. when you increase lambda_act in the ActivityConstraint to extreme values, at some point the cells will also break - but this is probably not so much an "artefact" as just an unrealistic parameter regime for the scenario you are trying to model. 

2. **Adjust your initial/seeding conditions**. 
    If you seed a cell in a single pixel, it starts out as much smaller than its target volume. This results in very large forces in the initial phase while the cell expands to attain that target volume. Especially if you seed many cells closely together, they may break apart due to these large forces; this is no real biological phenomenon but an artefact of the simulation (a real cell would not burst into existence from a single pixel). To fix this, you can try:
    - _Seeding cells not randomly but in predefined positions, so that each cell has the space to expand_. 
    (see e.g. this example: [https://artistoo.net/examples/Epidermis.html](https://artistoo.net/examples/Epidermis.html); from your browser you should be able to right click and select "view page source" to see the code)
    - _Seeding cells randomly, but giving each cell its own “burnin” period to let it expand before adding a new cell somewhere_. 
    In your code, you would have something like this:
        ```
        for( let i = 0; i < nCells; i++ ){
           sim.gm.seedCell( 1 )
           for( let t = 0; t < cellBurnin; t++ ) sim.C.timeStep()
        }
        ```
        Note that this is slow for large numbers of cells.
    - _Seed cells at a larger volume instead of just one pixel_.
    See https://artistoo.net/class/src/grid/GridManipulator.js~GridManipulator.html#instance-method-makeCircle.
    - _Temporarily change parameters during seeding_.
    If you use constraints like the ActivityConstraint, the positive feedback exacerbates the seeding problem because it makes the forces during volume expansion even bigger. So what you can do is temporarily set lambda_act to zero during the burnin period, and reset lambda_act to its original value once cells have attained their shape.
    - _Check that the number of cells you are seeding is not much larger than the space you are seeding them in_. 
    (so if ncells * target volume << total pixels on grid, you may run into problems).

3. **Phenomenologically prohibit cell breaking with a connectivity constraint**.
A ConnectivityConstraint simply forbids (or strongly discourages) any copy attempts that break the cell. See the different versions:
    - https://artistoo.net/class/src/hamiltonian/ConnectivityConstraint.js~ConnectivityConstraint.html
    - https://artistoo.net/class/src/hamiltonian/LocalConnectivityConstraint.js~LocalConnectivityConstraint.html
    - https://artistoo.net/class/src/hamiltonian/SoftConnectivityConstraint.js~SoftConnectivityConstraint.html
    - https://artistoo.net/class/src/hamiltonian/SoftLocalConnectivityConstraint.js~SoftLocalConnectivityConstraint.html

    I typically don’t like this “brute force” solution because it is not really linked to any physical process and can give weird artefacts in some cases. I therefore tend to focus on solutions 1 and 2 above, but this is an option if nothing else works.
 

Finally, to quantify cell breaking, we have previously used a metric called “connectedness”; see the methods in the paper https://www.sciencedirect.com/science/article/pii/S0006349521004240) and the artistoo implementation ( https://artistoo.net/class/src/stats/Connectedness.js~Connectedness.html). This may help if you are doing many simulations or even parameter screens and want to check that cells are not breaking too often.

Hope this helps!
